src/test/java/bio/terra/service/profile/ProfileAPIControllerTest.java

```diff
@@ -141,44 +150,66 @@ public void testUpdateProfileForbidden() {
     verify(profileService, times(0)).updateProfile(eq(billingProfileUpdateModel), eq(user));
   }
 
-  @Test
-  public void testDeleteProfile() {
+  @ParameterizedTest
+  @MethodSource
+  void testDeleteProfile(boolean deleteCloudResources, int expectedAdminAuthNumberOfInvocations) {
     when(authenticatedUserRequestFactory.from(any())).thenReturn(user);
     UUID deleteId = UUID.fromString("aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee");
     String jobId = "jobId";
-    when(profileService.deleteProfile(eq(deleteId), eq(user))).thenReturn(jobId);
+    when(profileService.deleteProfile(eq(deleteId), eq(deleteCloudResources), eq(user)))
+        .thenReturn(jobId);
+    when(profileService.getProfileByIdNoCheck(deleteId))
+        .thenReturn(new BillingProfileModel().id(deleteId));
+    doNothing().when(iamService).verifyAuthorization(any(), any(), any(), any());
 
     var jobModel = new JobModel();
     jobModel.setJobStatus(JobStatusEnum.RUNNING);
     when(jobService.retrieveJob(eq(jobId), eq(user))).thenReturn(jobModel);
 
-    ResponseEntity entity = apiController.deleteProfile(deleteId);
-    verify(profileService, times(1)).deleteProfile(eq(deleteId), eq(user));
+    ResponseEntity entity = apiController.deleteProfile(deleteId, deleteCloudResources);
+    // Only check for admin auth if deleteCloudResources is true
+    verify(iamService, times(expectedAdminAuthNumberOfInvocations))
+        .verifyAuthorization(
+            eq(user), eq(IamResourceType.DATAREPO), any(), eq(IamAction.CONFIGURE));
+    // Always check that the user has access on the spend profile
+    verify(iamService, times(1))
+        .verifyAuthorization(
+            eq(user),
+            eq(IamResourceType.SPEND_PROFILE),
+            eq(deleteId.toString()),
+            eq(IamAction.DELETE));
```

Comment:
As before, the arguments are all literals so `eq()` is not needed. Also `times(1)` is the default so that can be omitted.
```suggestion
    verify(iamService).verifyAuthorization(user,IamResourceType.SPEND_PROFILE, deleteId.toString(), IamAction.DELETE);
```

Reply:
Oh, ok, all the arguments must be constant to remove it. Thanks for that clarification!